chunky/src/java/se/llbit/chunky/main/Chunky.java

```diff
@@ -405,23 +395,35 @@ public SceneFactory getSceneFactory() {
   }
 
   @PluginApi
-  public void setPreviewRayTracerFactory(RayTracerFactory previewRayTracerFactory) {
-    this.previewRayTracerFactory = previewRayTracerFactory;
+  public void addPreviewRenderer(String name, Renderer renderer) {
+    if (InternalRenderManager.previewRenderers.containsKey(name))
+      Log.error("Preview renderer already exists (do you have the same plugin loaded twice?): " + name);
+    InternalRenderManager.previewRenderers.computeIfAbsent(name, n -> renderer);
   }
 
   @PluginApi
-  public RayTracerFactory getPreviewRayTracerFactory() {
-    return previewRayTracerFactory;
+  public void addRenderer(String name, Renderer renderer) {
+    if (InternalRenderManager.renderers.containsKey(name))
+      Log.error("Renderer already exists (do you have the same plugin loaded twice?): " + name);
+    InternalRenderManager.renderers.computeIfAbsent(name, n -> renderer);
   }
 
   @PluginApi
-  public void setRayTracerFactory(RayTracerFactory rayTracerFactory) {
-    this.rayTracerFactory = rayTracerFactory;
+  @Deprecated
+  public void setPreviewRayTracerFactory(RayTracerFactory previewRayTracerFactory) {
+    if (InternalRenderManager.previewRenderers.containsKey("Plugin Preview Renderer"))
+      Log.error("Attempted to register 2 plugin preview renderers.");
+    InternalRenderManager.previewRenderers.computeIfAbsent("Plugin Preview Renderer", name ->
+        new PreviewRenderer(previewRayTracerFactory.newRayTracer()));
```

Comment:
Just overwrite the previous renderer and log a warning. That would keep the previous behavior (last loaded plugin wins).